test/compress/collapse_vars.js

```diff
@@ -1209,3 +1209,24 @@ collapse_vars_regexp: {
         })();
     }
 }
+
+keep_collapse_const_in_own_block_scope: {
+    options = {
+        join_vars: true,
+        loops: true
+    }
+    input: {
+        var i=2;
+        const c=5;
+        while(i--)
+            console.log(i);
+        console.log(c);
+    }
+    expect: {
+        var i=2;
+        const c=5;
+        for(;i--;)
+            console.log(i);
+        console.log(c);
+    }
+}
```

Comment:
Hmm... I missed this comment... it's about var not being moved to the loop in all cases.